google/cloud/storage/internal/grpc_object_request_parser_test.cc

```diff
@@ -305,6 +306,58 @@ TEST(GrpcObjectRequestParser, InsertObjectMediaRequestWithObjectMetadata) {
   EXPECT_THAT(actual, IsProtoEqual(expected));
 }
 
+TEST(GrpcObjectRequestParser, ListObjectsRequestAllFields) {
+  google::storage::v2::ListObjectsRequest expected;
+  ASSERT_TRUE(google::protobuf::TextFormat::ParseFromString(
+      R"pb(
+        parent: "projects/_/buckets/test-bucket"
+        page_size: 10
+        page_token: "test-only-invalid"
+        delimiter: "/"
+        include_trailing_delimiter: true
+        prefix: "test/prefix"
+        versions: true
+        lexicographic_start: "test/prefix/a"
+        lexicographic_end: "test/prefix/abc"
+        common_request_params: { user_project: "test-user-project" }
+      )pb",
+      &expected));
+
+  ListObjectsRequest req("test-bucket");
+  req.set_page_token("test-only-invalid");
+  req.set_multiple_options(
+      MaxResults(10), Delimiter("/"), IncludeTrailingDelimiter(true),
+      Prefix("test/prefix"), Versions(true), StartOffset("test/prefix/a"),
+      EndOffset("test/prefix/abc"), UserProject("test-user-project"),
+      QuotaUser("test-quota-user"), UserIp("test-user-ip"));
```

Comment:
`QuotaUser` and `UserIp` are not used... but I see them unused in other tests, so there is probably a reason they are here.

Reply:
Ack, yes, they are intentional (mostly checking for "no crashes and no unexpected changes due to these extra options".